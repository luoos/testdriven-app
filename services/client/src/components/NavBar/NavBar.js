import React from 'react';
import { Link } from 'react-router-dom';

const NavBar = (props) => (
  <nav className="navbar is-dark has-shadow" role="navigation" aria-label="main navigation">
    <div className="container">
      <div className="navbar-brand">
        <strong className="navbar-item">TestDriven.io</strong>
        <span
          className="nav-toggle navbar-burger"
          onClick={() => {
            let toggle = document.querySelector(".nav-toggle");
            let menu = document.querySelector(".navbar-menu");
            toggle.classList.toggle("is-active");
            menu.classList.toggle("is-active");
          }}>
          <span></span>
          <span></span>
          <span></span>
        </span>
      </div>
      <div className="navbar-menu">
        <div className="navbar-start">
          <Link to="/" className="navbar-item">Home</Link>
          <Link to="/about" className="navbar-item">About</Link>
        </div>
        <div className="navbar-end">
            {props.isAuthenticated ?
              <div className="navbar-item has-dropdown is-hoverable">
                <a className="navbar-link">
                  More
                </a>
                <div className="navbar-dropdown">
                  <Link to="/profile" className="navbar-item">Profile</Link>
                  <hr className="navbar-divider"></hr>
                  <Link to="/logout" className="navbar-item">Log Out</Link>
                </div>
              </div>
              :
              <div className="navbar-item">
                <div className="buttons">
                  <Link to="/register" className="button is-success is-outlined">Register</Link>
                  <Link to="/login" className="button is-white">Log In</Link>
                </div>
              </div>
            }
        </div>
      </div>
    </div>
  </nav>
);

export default NavBar;