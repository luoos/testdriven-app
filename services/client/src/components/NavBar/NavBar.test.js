import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import NavBar from './NavBar';

describe('<NavBar />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });

  it('NavBar renders properly', () => {
    const element = wrapper.find('strong');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children).toBe('TestDriven.io')
  });

  it('Three navbar-start elements', () => {
    const element = wrapper.find('.navbar-start');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children.length).toBe(2);
  });

  it('NavBar renders a snapshot properly', () => {
    const tree = renderer.create(
      <MemoryRouter location="/"><NavBar /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})