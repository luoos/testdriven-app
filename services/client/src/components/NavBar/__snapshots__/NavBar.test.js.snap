// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> NavBar renders a snapshot properly 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-dark has-shadow"
  role="navigation"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <strong
        className="navbar-item"
      >
        TestDriven.io
      </strong>
      <span
        className="nav-toggle navbar-burger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </span>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-start"
      >
        <a
          className="navbar-item"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          className="navbar-item"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          className="navbar-item"
          href="/create"
          onClick={[Function]}
        >
          Create
        </a>
      </div>
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <div
            className="buttons"
          >
            <a
              className="button is-success is-outlined"
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
            <a
              className="button is-white"
              href="/login"
              onClick={[Function]}
            >
              Log In
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
